components/scream/src/diagnostics/field_at_single_pressure.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef EAMXX_FIELD_AT_SINGLE_PRESSURE_HPP
+#define EAMXX_FIELD_AT_SINGLE_PRESSURE_HPP
+
+#include "share/atm_process/atmosphere_diagnostic.hpp"
+#include "share/util/scream_vertical_interpolation.hpp"
+
+namespace scream
+{
+
+/*
+ * This diagnostic will produce the potential temperature.
+ */
+
+class FieldAtSinglePressure : public AtmosphereDiagnostic
+{
+public:
+  using Pack = ekat::Pack<Real,SCREAM_PACK_SIZE>;
+
+  // Constructors
+  FieldAtSinglePressure (const ekat::Comm& comm, const ekat::ParameterList& params);
+
+  // The name of the diagnostic
+  std::string name () const { return m_field_name + " @ pressure " + std::to_string(m_pressure_level) + " hPa"; }
+
+  // Set the grid
+  void set_grids (const std::shared_ptr<const GridsManager> grids_manager);
+
+protected:
+#ifdef KOKKOS_ENABLE_CUDA
+public:
+#endif
+  void compute_diagnostic_impl ();
+protected:
+
+
+
+  // Keep track of field dimensions
+  std::string         m_field_name;
+  FieldLayout         m_field_layout;
+  ekat::units::Units  m_field_units;
+  std::string         m_pres_name;
+
+  int                 m_pressure_level;
```

Comment:
This is supposed to be the value of the pressure, and not the level index, right? I understand we will not use decimal pressures, but I think it might be best if you use `Real` here. It doesn't change much, but helps disambiguate from a level index.